_posts/2024-02-14-provisioning-ubuntu-autoinstall.md

```diff
@@ -0,0 +1,361 @@
+- Because the ISO image is also required for the deployment but Foreman does not provide any dedicated interface for this, we have documented a way to place the ISO image into the `FQDN/pub/installation_media` directory on Foreman Server.
+This means Foreman serves the ISO image over `http`.
+If you want to deploy a host from a Smart Proxy Server, you have to copy the ISO image to your Smart Proxy Server.
+- Ubuntu Autoinstall deployments also require the extracted ISO image.
+You have to place the extracted ISO image in `FQDN/pub/installation_media/` on your Foreman Server and name the directory identical to the ISO image without the `.iso` file ending.
+If you want to deploy a host from a Smart Proxy Server, you have to copy the extracted ISO image to your Smart Proxy Server.
+
+Ensure that you provide the ISO and extracted ISO under the same name, for example `22_04.iso` and `22_04/`, to use the URL for the installation media as path for both.
+For more information, see [Creating an Installation Medium for Ubuntu 22.04](https://docs.theforeman.org/3.9/Provisioning_Hosts/index-katello.html#Creating_an_Installation_Medium_for_Ubuntu_22_04_provisioning) in _Provisioning Hosts_.
+
+#### preseed_kernel_options_autoinstall.erb
+
+```erb
+<%#
+kind: snippet
+name: preseed_kernel_options_autoinstall
+model: ProvisioningTemplate
+snippet: true
+description: options for the kernel / preseed startup initialization
+oses:
+- ubuntu
+test_on:
+- ubuntu_autoinst4dhcp
+-%>
+<%
+  ...
+  image_path = @preseed_path.sub(/\/?$/, '.iso')
+  userdata_option = "ds=nocloud-net;s=http://#{foreman_request_addr}/userdata/"
+  options = []
+
+  ...
+
+  options << "BOOTIF=#{mac}" if mac
+  options << 'ramdisk_size=1500000'
+  options << 'fsck.mode=skip'
+  options << 'autoinstall'
+  options << "url=http://#{@preseed_server}#{image_path}"
+  options << 'cloud-config-url=/dev/null'
+  if @add_userdata_quotes
+    options << "\"#{userdata_option}\""
+  else
+    options << userdata_option
+  end
+  options << 'console-setup/ask_detect=false'
+  options << "locale=#{host_param('lang') || 'en_US'}"
+  options << 'localechooser/translation/warn-light=true'
+  options << 'localechooser/translation/warn-severe=true'
+  options << "hostname=#{hostname}"
+  options << "domain=#{domain}"
+%>
+<%= options.join(' ') -%>
+```
+
+For more information, see [app/views/unattended/provisioning_templates/snippet/preseed_kernel_options_autoinstall.erb](https://github.com/theforeman/foreman/blob/develop/app/views/unattended/provisioning_templates/snippet/preseed_kernel_options_autoinstall.erb) in the _foreman_ repository.
+* To complete the provisioning process, the user data template configures a user, sets up remote access using authorized SSH keys, configures networking and the disk layout, and fetches the rendered finish data template from Foreman.
+The `Preseed default finish` template installs `subscription-manager` and registers the host to Foreman.
```

Comment:
Why is this a list with a single entry?

Reply:
Leftover from switching from asciidoc to markdown :see_no_evil: Dropped.